bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -0,0 +1,339 @@
+
+        protected Integer getValue() {
+            return intValue;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("%s (%d)", text, intValue);
+        }
+    }
+
+    /**
+     * Describes a shade type entry in the database; implements 'capabilities' parameter.
+     *
+     * @author Andrew Fiddian-Green - Initial Contribution
+     */
+    public static class Type extends Base {
+        private int capabilities = -1;
+
+        protected Type() {
+        }
+
+        protected Type(int type) {
+            intValue = type;
+        }
+
+        protected Type text(String text) {
+            this.text = text;
+            return this;
+        }
+
+        protected Type capabilities(int capabilities) {
+            this.capabilities = capabilities;
+            return this;
+        }
+
+        /**
+         * Get shade types's 'capabilities'.
+         *
+         * @return 'capabilities'.
+         */
+        public int getCapabilities() {
+            return capabilities;
+        }
+    }
+
+    /**
+     * Describes a shade 'capabilities' entry in the database; adds properties indicating its supported functionality.
+     *
+     * @author Andrew Fiddian-Green - Initial Contribution
+     */
+    public static class Capabilities extends Base {
+        private boolean supportsPrimary;
+        private boolean supportsSecondary;
+        private boolean supportsTiltOnClosed;
+        private boolean supportsTiltAnywhere;
+        private boolean primaryStateInverted;
+        private boolean tilt180Degrees;
+
+        public Capabilities() {
+        }
+
+        protected Capabilities(int capabilities) {
+            intValue = capabilities;
+        }
+
+        protected Capabilities text(String text) {
+            this.text = text;
+            return this;
+        }
+
+        protected Capabilities primary() {
+            supportsPrimary = true;
+            return this;
+        }
+
+        protected Capabilities tiltOnClosed() {
+            supportsTiltOnClosed = true;
+            return this;
+        }
+
+        protected Capabilities secondary() {
+            supportsSecondary = true;
+            return this;
+        }
+
+        protected Capabilities tiltAnywhere() {
+            supportsTiltAnywhere = true;
+            return this;
+        }
+
+        protected Capabilities primaryStateInverted() {
+            primaryStateInverted = true;
+            return this;
+        }
+
+        protected Capabilities tilt180() {
+            tilt180Degrees = true;
+            return this;
+        }
+
+        /**
+         * Check if the Capabilities class instance supports a primary shade.
+         *
+         * @return true if it supports a primary shade.
+         */
+        public boolean supportsPrimary() {
+            return supportsPrimary;
+        }
+
+        /**
+         * Check if the Capabilities class instance supports a vane.
```

Comment:
Method description mentions vane support, but name mentions tilt anywhere support. Is it the same? My Twist (44) with capabilities "Bottom Up (0)" has a vane/slat, but property is: "tiltAnywhereDetected: false"

Reply:
A couple of comments..
- In general 'Vane' and 'Tilt' mean the same function -- namely to adjust the light transmission without changing the vertical position; albeit perhaps making a different physical movement on different types of shade.
- There are two types of vane/tilt -- namely a) the shade has one motor that influences both the position **or** vane/tilt, and b) the shade has two motors that can fully independently move the position **and** vane/tilt. The former strictly can only adjust the vane/tilt when the (e.g. venetian) blind has been fully opened; and any movement after it is open, adjusts the vanes; typically they only have 90 degrees of vane/tilt (dark .. light). The latter can literally move the vane/tilt at any shade position; and typically they have 180 degrees of vane/tilt (dark .. light .. dark). The former type of vane/tilt is included in `capabilities:0` whereas `tiltAnywhere` is declared in another class of `capabilities`.
- And yes the Twist blinds are indded "special" :) (I have some, so I know them well). They have just one motor, so they are declared as `capabilities:0` i.e. NOT supporting tilt anywhere. But the oddity of Twist is that you can indeed adjust the overlap of the bars (dark .. light .. dark) at any shade position by means of its single motor. Nevertheless electronically the hub reports only a `position1` value that can be either `posKind:1` (shade) OR `posKind:3` (vane/tilt) [to be specific it does NOT report a separate `position2`]. And the interpretation of the `position1` / `posKind:1` / `posKind:3` values is indeed as you describe in your own post below, and indeed as described in the ReadMe.

However I will update the comments to eliminate any confusion over vane/tilt.

Reply:
Thanks for the detailed explanation - and for the clarification in the method doc.